src/frontend/src/views/Data/components/TreesView/components/TreesTable/components/TreeActionMenu/components/MoreActionsMenu/index.tsx

```diff
@@ -51,10 +68,40 @@ const MoreActionsMenu = ({
     setAnchorEl(null);
   };
 
+  const handleOpenEditModal = () => {
+    if (usesTableRefactor) {
+      setIsEditTreeModalOpen(true);
+    } else {
+      onEditTreeModalOpen && onEditTreeModalOpen(item);
+    }
+  };
+
+  const handleOpenDeleteModal = () => {
+    if (usesTableRefactor) {
+      setIsDeleteTreeModalOpen(true);
+    } else {
+      onDeleteTreeModalOpen && onDeleteTreeModalOpen(item);
+    }
+  };
+
   const open = Boolean(anchorEl);
 
   return (
     <>
+      {usesTableRefactor && (
+        <>
+          <EditTreeConfirmationModal
+            open={isEditTreeModalOpen}
+            phyloRun={item}
+            onClose={() => setIsEditTreeModalOpen(false)}
```

Comment:
I think moving this from an inline arrow function to a not-inline function makes sense for these.  We have a row for every tree and afaik we aren't paginating.  Seems like an easy (if small) performance win.

Reply:
- [ ] move to not inline function

Reply:
Is this actually a performance win? I'm not clear on why. Even if I define the handlers as variables, wouldn't they still be redefined on every re-render?

Reply:
I'm fairly certain it must be the case that variables are redefined on every re-render, at least for functional components (almost all of ours are). If not, then variables like `const shouldShowTooltip = Boolean(tooltipProps);` wouldn't hold different values on any render after the first one, even if the props changed. 

The reason I chose to inline these is because I feel it's more readable when the callback is just one line, rather than defining variables up top and then inlining them. Though I do prefer the vars if the callback is any more than just one line.

Reply:
I imagine this is a big part of why `useState` is a thing. If variables were not reset every render, then you could use a regular variable to hold component state. See [this](https://beta.reactjs.org/learn/state-a-components-memory#when-a-regular-variable-isnt-enough) `handleClick` example.

Reply:
I'm convinced.  I also find them more readable.

Reply:
I agree with where this conversation landed. I don't think we need to worry about the function being created in-line every render.

That said, if it was a different situation and we really did need a way to avoid re-creating the function on every re-render, I think `useCallback` is what you'd want to reach for: https://reactjs.org/docs/hooks-reference.html#usecallback

My understanding is:
- `useMemo` -- You have a function that is expensive to run, but its output stays (mostly) the same between re-renders, so you want to only run the function when its inputs change, not on every single render. `useMemo` lets you capture that value and only re-run if an input changes, regardless of renders happening.
- `useCallback` -- You have a function that is expensive to create -- say, you have a function factory that outputs the function, but that factory is expensive to run -- or when keeping reference equality for the function between renders is helpful for child components. `useCallback` lets you capture that function and hold on to it so it doesn't need re-creation every re-render. It's basically a `useMemo`, but for functions (which the React docs explicitly call out).